x/oracle/keeper/band_oracle_test.go

```diff
@@ -132,3 +133,77 @@ func TestBandCallDataRecord(t *testing.T) {
 	record = app.OracleKeeper.GetBandCallDataRecord(ctx, 1)
 	require.Nil(t, record)
 }
+
+func TestGetPrice(t *testing.T) {
+	app := app.Setup(t, false)
+	ctx := app.BaseApp.NewContextLegacy(false, tmproto.Header{Height: 1, ChainID: "3", Time: time.Unix(1618997040, 0)})
+
+	// Case 1: Base, quote price, does not exist, expect nil
+	price := app.OracleKeeper.GetPrice(ctx, "ATOM", "USD")
+	require.Nil(t, price)
+
+	// Case 2: Set base price state for ATOM
+	bandPriceStateATOM := &types.BandPriceState{
+		Symbol:      "ATOM",
+		Rate:        math.NewInt(10),
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(10), 1),
+	}
+	err := app.OracleKeeper.SetBandPriceState(ctx, "ATOM", bandPriceStateATOM)
+	require.NoError(t, err)
+
+	// // Case 3: Set quote price state for USD
+	bandPriceStateUSD := &types.BandPriceState{
+		Symbol:      "USD",
+		Rate:        math.NewInt(1),
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(1), 1),
+	}
+	err = app.OracleKeeper.SetBandPriceState(ctx, "USD", bandPriceStateUSD)
+	require.NoError(t, err)
+
+	// Case 4: Base price is invalid (rate is zero), expect nil
+	invalidPriceState := &types.BandPriceState{
+		Symbol:      "ATOM",
+		Rate:        math.NewInt(0), // Invalid base rate
+		ResolveTime: 1,
+		Request_ID:  1,
+		PriceState:  *types.NewPriceState(math.LegacyNewDec(0), 1),
+	}
+	err = app.OracleKeeper.SetBandPriceState(ctx, "ATOM", invalidPriceState)
+	require.NoError(t, err)
+
+	price = app.OracleKeeper.GetPrice(ctx, "ATOM", "USD")
+	require.Nil(t, price)
+
+	// Case 5: Quote price is invalid (rate is zero), expect nil
```

Comment:
This case seems wrong because when `quote=USD` the code will return here https://github.com/onomyprotocol/reserve/blob/e4055d4d8311d2e6a03b1f1f5356916e1defbe81/x/oracle/keeper/band_oracle.go#L226  but because you set band price state for `ATOM` at line 175 above with invalid `rate` so maybe it return nil at here (we tested in case 4)https://github.com/onomyprotocol/reserve/blob/1230d5e435abda41e94759e81d6ea7113615cabb/x/oracle/keeper/band_oracle.go#L222
I recommend we should rewrite this test with table-driven test to easy keep track all the test case and follow the logic of each test case